### DISCUSSION TITLE
Should Metals work with plain Scala3 files directly in VSCode workspace? ...no build tool, like SBT, is used

### DISCUSSION THREAD

**Discussion by jockbert** [scalameta/metals#3336] [Q&A] [answered]
I'm trying out Metals for coding Scala 3 in VSCode. No build tool is used, so there is no project for Metals to import. Each Scala file is a separate stand-alone program that I run directly with `scala` command in terminal. like this:

```
$ scala day3.1.sc input3A.txt
... some output
$ scala --version
Scala code runner version 3.1.0 -- Copyright 2002-2021, LAMP/EPFL
```
However, Metals seems confused and complain about syntax that the `scala` command has no issues with. Further Metals also tries to download Ammonite 2.3.8-124-2da846d2 for scala 2.13.7 (!), but failed.  In an effort to solve the problem, I have tried to specify `runner.dialect = scala3` in the file `.scalafmt.conf`, as [mentioned here](https://scalameta.org/scalafmt/docs/configuration.html#scala-3), but it does not help. See screenshot for details:

![plain Scala3 files and Metals](https://user-images.githubusercontent.com/3252671/144727448-c725e0af-2fd1-43aa-ad78-975ea3b7563d.png)

What do I need to do in order to have a good experience trying out Scala 3 using Metals in VSCode? Besides problem with the Scala 3 syntax, Metals does not show any documentation for methods in standard lib. Documentation for my own code works fine though.

**Comment by ckipp01** [comment, marked as answer]
Hey @jockbert so keep in mind that Metals is interpreting your `.sc` files as Ammonite scripts, not just normal Scala files. Because it's doing that it's trying to download the correct version of Ammonite. Metals doesn't yet support Ammonite with Scala 3, so those won't work. 

In order to get a better experience with Scala 3 I'd recommend either putting those in a project with a valid Scala 3 build, making them `.scala` instead of `.sc` if they can truly stand alone, or maybe even using https://scala-cli.virtuslab.org/ which then Metals could connect to for standalone scripts and support Scala 3.

**Reply by jockbert** [reply]
Thanks, renaming to from `.sc` to `.scala` seems to fix the issue with false Scala problems (red markings) in editor. It was just ignorance using the suffix `.sc`, since Ammonite is not even used for these files.
